doc: Adds upgrade guide for MongoDB Atlas Provider 1.32.0 with new features and resources (#3260)
mongodb/terraform-provider-mongodbatlas: docs/guides/1.32.0-upgrade-guide.md

---
page_title: "Upgrade Guide 1.32.0"
---

# MongoDB Atlas Provider 1.32.0: Upgrade and Information Guide

The Terraform MongoDB Atlas Provider version 1.32.0 has a number of new and exciting features.

> ## New Resources, Data Sources, and Features

Reviewer: There's also new description field in resource policy https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md#unreleased

Author: Thank you, added in 5abf40dd1500501336b39ec5433bfb87b4060cbc

> - You can now use encryption at rest for `mongodbatlas_search_deployment` by configuring the `mongodbatlas_encryption_at_rest` resource with `enabled_for_search_nodes`. Then verify the change with the new `encryption_at_rest_provider` attribute on the `mongodbatlas_search_deployment` resource. To learn more, see the [mongodbatlas_encryption_at_rest](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest#enabled_for_search_nodes-1) resource documentation.

Reviewer 2: have you checked anchor #enabled_for_search_nodes-1 is correct?

Author: Is it possible to test without generating the provider?
I assume it works since the only difference to existing attributes is the attribute name

Reviewer 2: links are correct, thanks

Author: Thank you for checking 👍

> - You can now use encryption at rest for `mongodbatlas_search_deployment` by configuring the `mongodbatlas_encryption_at_rest` resource with `enabled_for_search_nodes`. Then verify the change with the new `encryption_at_rest_provider` attribute on the `mongodbatlas_search_deployment` resource. To learn more, see the [mongodbatlas_encryption_at_rest](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest#enabled_for_search_nodes-1) resource documentation.

Reviewer 3: ```suggestion
- You can now enable encryption at rest for MongoDB Atlas Search Nodes by configuring the `mongodbatlas_encryption_at_rest` resource with the `enabled_for_search_nodes` attribute, which can be verified using the `encryption_at_rest_provider` attribute on the `mongodbatlas_search_deployment` resource. To learn more, see the [mongodbatlas_encryption_at_rest](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest#enabled_for_search_nodes-1) and the [mongodbatlas_search_deployment](LINK_HERE) resource documentation.
```

Reviewer 3: Should we also link to the `encryption_at_rest_provider` from the [search_deployment doc](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/317ca9b559ed1aec33c6f69acdf5c05ed9942b6b/docs/resources/search_deployment.md#L4)? Not sure what would be the link though

Author: The challenge is that the attribute where we could link is auto-generated based on the schema attribute description. Seems a bit strange to add a link to the encryption at rest (`https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest#enabled_for_search_nodes-1`) from the `searchdeployment/schema.go#encryption_at_rest_provider.description`.

Reviewer 3: I am not sure I follow. What part of my message are you referring to be challenging? the link to the `encryption_at_rest_provider` field?

Author: Added suggestion here: 5cfac562f97b4b2564e633368a0e6288153a933f

The Terraform MongoDB Atlas Provider version 1.32.0 has a number of new and exciting features.

## New Resources, Data Sources, and Features

- You can now enable encryption at rest for MongoDB Atlas Search Nodes by configuring the `mongodbatlas_encryption_at_rest` resource with the `enabled_for_search_nodes` attribute, which can be verified using the `encryption_at_rest_provider` attribute on the `mongodbatlas_search_deployment` resource. To learn more, see the [mongodbatlas_encryption_at_rest](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest#enabled_for_search_nodes-1) and the [mongodbatlas_search_deployment](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/search_deployment#encryption_at_rest_provider-1) resource documentations.

- `mongodbatlas_search_deployment` now supports `skip_wait_on_update` to avoid waiting for the update operations to complete. To learn more, see the [mongodbatlas_search_deployment](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/search_deployment) resource documentation.

> - You can now update the `mongodbatlas_stream_processor`. To learn more, see the [mongodbatlas_stream_processor](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/stream_processor) resource documentation.

Reviewer 3: ```suggestion
- You can now update the `mongodbatlas_stream_processor` resource. To learn more, see the [mongodbatlas_stream_processor](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/stream_processor) resource documentation.
```

Author: applied in 34f3d8122cc6e49c504dd722aa4eaadbd5837d81

> - `mongodbatlas_resource_policy` resource and data sources now supports the new `description` field. To learn more, see the [mongodbatlas_resource_policy](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/resource_policy#description-1) resource documentation.

Reviewer: ```suggestion
- `mongodbatlas_resource_policy` resource and data sources now support the new `description` field. To learn more, see the [mongodbatlas_resource_policy](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/resource_policy#description-1) resource documentation.
```

Author: Thank you. Applied in 34f3d8122cc6e49c504dd722aa4eaadbd5837d81
